instance/tests/integration/factories/instance.py

```diff
@@ -39,7 +39,16 @@ class OpenEdXInstanceFactory(DjangoModelFactory):
     class Meta:
         model = OpenEdXInstance
 
-    sub_domain = factory.LazyAttribute(lambda o: '{}.integration'.format(str(uuid.uuid4())[:8]))
+    @classmethod
+    def create(cls, *args, **kwargs):
+        # OpenEdXInstance constructor accepts either a 'sub_domain' or 'instance_lms_domain' value. Only generate a
+        # random value for 'internal_lms_domain' if neither 'sub_domain' nor 'internal_lms_domain' are provided.
+        if 'sub_domain' not in kwargs and 'internal_lms_domain' not in kwargs:
+            kwargs = kwargs.copy()
+            domain = generate_internal_lms_domain('{}.integration'.format(str(uuid.uuid4())[:8]))
```

Comment:
@mtyaka For readability I'd probably split this up into two lines:

```python
            sub_domain = '{}.integration'.format(str(uuid.uuid4())[:8])
            domain = generate_internal_lms_domain(sub_domain)
```

Or even:

```python
            random_id = str(uuid.uuid4())[:8]
            sub_domain = '{}.integration'.format(random_id)
            domain = generate_internal_lms_domain(sub_domain)
```

That way it's immediately obvious that `generate_internal_lms_domain` expects to receive a subdomain.